### PR TITLE
several updates to exo2nek

### PR DESCRIPTION
**Pull request by yhaomin2007**
1. update for super large mesh >100 million elements
2. offset side-set number to start from 1 consecutively
3. update readme for new features

### CODE REVIEW

File: tools/exo2nek/exo2nek.f90 [hunk 1]
```diff
@@ -6,23 +6,23 @@
 ! 4, setting periodicity
 ! 5, tet-to-hex, wedge-to-hex conversion
 !---------------------------------------------------------------------------------
+! Feb 2021,
```
Comment (stgeke) [inline tools/exo2nek/exo2nek.f90]: Please remove these comments

File: tools/exo2nek/exo2nek.f90 [hunk 2]
```diff
@@ -6,23 +6,23 @@
...
-      !logical ifmoresolid
 
-      write(6,*) 'Special exo2nek version to merge multiple fluid/solid exo files'
-      write(6,*) 'notes: for 3D mesh only'
-      write(6,*) 'notes: interface surface mesh should be conformal and match spatially'
+      write(6,*) 'exo2nek version to merge multiple fluid/solid exo files'
+      write(6,*) 'notes: 3D mesh only'
+      write(6,*) 'notes: interface surface mesh should be conformal'
 
       write(6,*) 'please input estimate final total hex element number:'
```
Comment (stgeke) [inline tools/exo2nek/exo2nek.f90]: Why is this input required?

File: tools/exo2nek/exo2nek.f90 [hunk 3]
```diff
@@ -3050,7 +3150,7 @@ subroutine exitt
 !-----------------------------------------------------------------------
       subroutine copy(a,b,n)
       real a(1),b(1)
-
+      integer*8 n
```
Comment (stgeke) [inline tools/exo2nek/exo2nek.f90:2281]: Why is n integer*8 but i is not?

File: tools/exo2nek/exo2nek.f90 [hunk 4]
```diff
@@ -2484,28 +2573,31 @@ subroutine gather_bc_info
...
       end
 	  
 !--------------------------------------------------------------------
       subroutine scale_mesh
       use SIZE
-      real xx,yy,zz,ss
-      integer ntot
-
+      real*8 xx,yy,zz,ss
+      integer*8 ie
```
Comment (stgeke) [inline tools/exo2nek/exo2nek.f90]: I think it's good enough to be integer*4 - the converter will anyway not support more than say (2**32)/2-1 elements.  Please correct all the other locations as well.

File: tools/exo2nek/exo2nek.f90 [hunk 5]
```diff
@@ -6,23 +6,23 @@
...
       integer option
-      character ifmore,if_stitch_interface
-      logical ifmorefluid
       integer iexo1
-      !logical ifmoresolid
 
-      write(6,*) 'Special exo2nek version to merge multiple fluid/solid exo files'
-      write(6,*) 'notes: for 3D mesh only'
-      write(6,*) 'notes: interface surface mesh should be conformal and match spatially'
+      write(6,*) 'exo2nek version to merge multiple fluid/solid exo files'
```
Comment (stgeke) [inline tools/exo2nek/exo2nek.f90]: Please remove those comments. This is explained in the README already.

**Comment by yhaomin2007** [comment]
I changed exo2nek.f90 according to most of the comments. 
But the number "please input estimate final total hex element number" is needed here to allocate arrays.

if you want this process to be automatic, then I have to find a way to pre-read all exo files  and user options.
This should be ok, but it may take some time.

**Comment by stgeke** [comment]
That very bad from a UX perspective. Please eliminate this question.

**Comment by yhaomin2007** [comment]
ok, will do.

**Comment by yhaomin2007** [comment]
New interface was created. please review. thanks.

**Pull request merged**